tutorials/shaders/shader_reference/shader_functions.rst

```diff
@@ -0,0 +1,3804 @@
+.. _doc_shader_functions:
+
+Built-in functions
+==================
+
+Godot supports a large number of built-in functions, conforming roughly to the
+GLSL 4.00 specification.
```

Comment:
The [shading language](https://docs.godotengine.org/en/stable/tutorials/shaders/shader_reference/shading_language.html#introduction) page says godot's shading language is based on `GLSL ES 3.0`. So one of these two pages is wrong. (Or for built-in functions, there is no difference between 3.0 and 4.0)

Reply:
I need to go some digging into Godot's source to try to find the answer because I saw that too, but some of these functions weren't listed in the opengl es 3.0 documentation I was originally looking in. So need to find a definitive source of truth for which version is supported.

Reply:
If you blame the shading language page, that phrasing was added in [this commit ](https://github.com/godotengine/godot-docs/commit/ff696205cea64ca6b624c918c53fdd90ba4e2482). It's only had minor phrasing and formatting changes since then.

Reply:
Oh, lol. Committed 7 years ago. Makes me think very strongly that the info is just outdated.

Now I'm a little worried that the docs and the code have diverged a bit. Maybe I should compare the functions listed in docs to those actually supported by code.

Reply:
An easy solution would be to only specify "based on GLSL" without the version and link to the current version of the GLSL spec.
Ideally we would coordinate the change in wording across this page, the Shading Language page, and the migrating from GLSL page.
For this PR I think you just need to make sure that you're not *adding* wrong information, the other cases can be changed separately.

Reply:
Well, trolled the code and other things for a while before I realized that I was probably confusing OpenGL ES shader language (commonly ESSL or GLSL ES) and OpenGL Shader Language (GLSL). Turns out there isn't an ESSL 4.5, so I must have looked at GLSL documentation .

Wikipedia says that the two languages are related, but not compatible, so that's fun. So yeah, there's quite a few references in the engine code to GLES3, though there's also a lot of glsl files that reference "#version 450" which is definitely a GLSL version. So there seems to be substantial differences in shader compilation for Vulkan vs. D3D12 vs Legacy OpenGL backends. Guess that's why Godot needs its own shader language.

So yeah, I definitely need to change that statement.

Reply:
Ok, pushed a commit that fixed that wording, and the couple formatting issues you specifically mentioned above. I'll probably leave the rest of the formatting issues for you.